3rdparty/CMakeLists.txt

```diff
@@ -578,8 +583,17 @@ function(NBL_ADJUST_FOLDER NBL_TARGET NBL_PREFIX KEEP_PRESENT_PREFIX)
 	endif()
 endfunction()
 
-NBL_GET_ALL_TARGETS(NBL_ALL_TARGETS)
-foreach(NBL_TARGET IN LISTS NBL_ALL_TARGETS)
+NBL_GET_ALL_TARGETS(NBL_ALL_3RDPARTY_TARGETS)
+foreach(NBL_TARGET IN LISTS NBL_ALL_3RDPARTY_TARGETS)
+	get_target_property(NBL_TARGET_SOURCE_DIR ${NBL_TARGET} SOURCE_DIR)
+	if(${NBL_TARGET_SOURCE_DIR} MATCHES "${NBL_ROOT_PATH}/3rdparty/dxc*" OR ${NBL_TARGET_SOURCE_DIR} MATCHES "${NBL_ROOT_PATH}/3rdparty/libjpeg-turbo*")
```

Comment:
have a `NBL_3RDPARTY_IMMUTABLE_TARGET_SRC_PATTERNS` list, put `"${NBL_ROOT_PATH}/3rdparty/dxc*"` and `"${NBL_ROOT_PATH}/3rdparty/libjpeg-turbo*"` into the list, & use it to perform the regex pattern check

Reply:
```cmake
# you can move this util to common.cmake
function(NBL_MATCH_PATTERNS IN_STRING PATTERNS_VAR RESULT_VAR)
    set(NBL_MATCHED FALSE)

    foreach(PATTERN IN LISTS ${PATTERNS_VAR})
        if(${IN_STRING} MATCHES "${PATTERN}")
            set(NBL_MATCHED TRUE)
            break()
        endif()
    endforeach()

    set(${RESULT_VAR} ${NBL_MATCHED} PARENT_SCOPE)
endfunction()

set(NBL_3RDPARTY_IMMUTABLE_TARGET_SRC_PATTERNS
    "${NBL_ROOT_PATH}/3rdparty/dxc*"
    "${NBL_ROOT_PATH}/3rdparty/libjpeg-turbo*"
)

NBL_GET_ALL_TARGETS(NBL_ALL_3RDPARTY_TARGETS)

foreach(NBL_TARGET IN LISTS NBL_ALL_3RDPARTY_TARGETS)
    get_target_property(NBL_TARGET_SOURCE_DIR ${NBL_TARGET} SOURCE_DIR)
    NBL_MATCH_PATTERNS("${NBL_TARGET_SOURCE_DIR}" NBL_3RDPARTY_IMMUTABLE_TARGET_SRC_PATTERNS MATCH_FOUND)

    # PS. you can try to use this trick with generator expression and make it one-liner instead of if statement
    if(MATCH_FOUND)
        list(APPEND NBL_IMMUTABLE_TARGETS ${NBL_TARGET})
    else()
        list(APPEND NBL_MUTABLE_TARGETS ${NBL_TARGET})
    endif()
endforeach()
```

Reply:
by the way why do we treat `libjpeg-turbo` as immutable?